### DISCUSSION TITLE
Babel com npm scripts

### DISCUSSION THREAD

**Discussion by allysonsouza** [frontendbr/forum#1042] [Perguntas e Respostas]
Na issue #746 eu perguntei sobre WebPack para incluir no meu workflow WordPress, porém no final optei por ir de npm scripts mesmo.

Agora estou tentando aprimorar o meu npm scritps para utilizar o Babel e poder usar os últimos recursos do JavaScript em meus projetos, porém estou apanhando bonito para configurar, fora que me surgiram várias dúvidas. O contexto é aplicação em tema WordPress que utiliza Bootstrap 4, o qual instalo via npm e preciso carregar junto com a popper.js (requisito do Bootstrap) e jQuery, e também gostaria de escolher os módulos do Bootstrap utilizado para um bundle mais leve no final.

1. Preciso do Browserify para gerenciar o carregamento de módulos?
2. Qual a sequência dos processo que devo realizar? Browserify > Babel > Uglify? Ou o Babel já roda um uglify?
3. Devo largar mão do npm scripts e ir, sei lá, pro Gulp e minha vida será muito mais fácil? xD (Foi tranquilo configurar npm scripts para as besteirinhas que preciso em um tema WordPress, o problema tá sendo o Babel)

Obrigado desde já!

**Comment by allysonsouza** [comment]
Ah sim, `package.json` atual aqui: https://github.com/HasteDesign/Haste-Starter/blob/master/package.json

**Comment by ninetails** [comment]
**TL;DR** "Devo largar mão do npm scripts e ir, sei lá, pro Gulp e minha vida será muito mais fácil?" - SIM

Os gerenciadores de bundler eles estão ae pra automatizar todos esses npm scripts que vc fez separadamente. Se for mesmo utilizar o Gulp, precisará sim do Browserify pois ele é quem pegará os módulos na pasta node_modules e injetará no seu bundle.

Mas eu recomendaria ao invés de ir pro Gulp, ver sobre o Parcel, Rollup ou o Webpack. Parcel pela filosofia zero config, mas se tiver algo que não gostar ou achar que falta algo (não sei se dará pra utilizar PostCSS por exemplo), os outros dois farão mais sentido mas de qualquer jeito precisará dedicar um tempo pra configurar.
